services/bonita-business-data/bonita-business-data-client-resources/src/test/java/org/bonitasoft/engine/bdm/dao/client/resources/utils/EntityGetterTest.java

```diff
@@ -26,88 +26,121 @@
     public void can_only_be_applicable_to_getters() throws Exception {
+        //given
         Method setter = Employee.class.getMethod("setManager", Employee.class);
+
+        //when then
         new EntityGetter(setter);
     }
 
     @Test
     public void should_be_able_to_retrieve_source_entity_name() throws Exception {
+        //given
         Method methodOfEmployee = Employee.class.getMethod("getManager");
 
+        //when
         String sourceEntityName = new EntityGetter(methodOfEmployee).getSourceEntityName();
 
+        //then
         assertThat(sourceEntityName).isEqualTo(Employee.class.getSimpleName());
     }
 
     @Test
     public void should_be_able_to_retrieve_target_entity_class() throws Exception {
+        //given
         Method methodReturningAdress = Employee.class.getMethod("getAddress");
 
+        //when
         Class<?> targetEntityClass = new EntityGetter(methodReturningAdress).getTargetEntityClass();
 
+        //then
         assertThat(targetEntityClass).isEqualTo(Address.class);
     }
 
     @Test
     public void should_be_able_to_retrieve_target_entity_class_even_if_method_return_a_list() throws Exception {
+        //given
         Method methodReturningListOfAdresses = Employee.class.getMethod("getAddresses");
 
+        //when
         Class<?> targetEntityClass = new EntityGetter(methodReturningListOfAdresses).getTargetEntityClass();
 
+        //then
         assertThat(targetEntityClass).isEqualTo(Address.class);
     }
 
     @Test
     public void should_be_able_to_retrieve_capitalized_field_name() throws Exception {
+        //given
         Method getAdresses = Employee.class.getMethod("getAddresses");
 
+        //when
         String sourceEntityName = new EntityGetter(getAdresses).getCapitalizedFieldName();
 
+        //then
         assertThat(sourceEntityName).isEqualTo("Addresses");
     }
 
     @Test
     public void should_return_entity_class_name_if_getter_return_a_unique_object() throws Exception {
+        //given
         Method uniqueObjectGetter = Employee.class.getMethod("getAddress");
 
+        //when
         String classname = new EntityGetter(uniqueObjectGetter).getReturnTypeClassName();
 
+        //then
         assertThat(classname).isEqualTo(Address.class.getName());
     }
 
     @Test
-    public void should_return_list_class_name_if_getter_return_multiple_objects() throws Exception {
+    public void should_return_list_object_class_name_if_getter_return_multiple_objects() throws Exception {
+        //given
         Method multipleObjectGetter = Employee.class.getMethod("getAddresses");
 
-        String classname = new EntityGetter(multipleObjectGetter).getReturnTypeClassName();
+        //when
+        String className = new EntityGetter(multipleObjectGetter).getReturnTypeClassName();
 
-        assertThat(classname).isEqualTo(List.class.getName());
+        //then
+        assertThat(className).isNotEqualTo(List.class.getName());
```

Comment:
may be this test can be merged with should_be_able_to_retrieve_target_entity_class_even_if_method_return_a_list